### PR TITLE
Company Details Accordion

### LINKED ISSUE #50: Company Details Accordion
## Description

Create a new accordion component for Company Details Description. We should be hiding the text after 5 lines. 

image

## Learning Objectives

- Tailwind CSS 
- React Components 

## Todo

These points are a rough guideline. Please feel free to discuss with others on the team about the best way to design this component!

- [ ] Create the accordion component (handles hiding text that is longer than 5 lines) 
- [ ] Pass in company description and render on company details page

## Useful Links
- [Figma Designs](https://www.figma.com/file/PFCGAeZzwLEyzNaovh6Zfq/Toriis?node-id=3-1346&t=SnMEgVsxUIiIsB6Y-0)

### PR DESCRIPTION
**Pull request by lizageorge**
## Summary

A new accordion for the company details page, limits text after 75 words (5 lines on full width).
Closes #50 

## Changes

- Created and styled CompanyDetailsAccordion component
- Add ReadMore Button
- Set display to change between full and truncated text based on ReadMore Button

## Screenshots

image
image

"Read More" button is hidden when starting task is short enough:
image

### CODE REVIEW

File: src/pages/company/[id].tsx [hunk 1]
```diff
@@ -45,6 +45,11 @@ const Company = () => {
       <div className="flex flex-col items-center">
         <HighlightedTitle title={data.name} size="large" color="clementine" />
       </div>
+
+      <CompanyDetailsAccordion
+        content={data.description ? data.description : ''}
+      />
```
Comment (zorazrr) [inline src/pages/company/[id].tsx:51]:
```suggestion
      {data.description && <CompanyDetailsAccordion content={data.description} />}
```

Comment (zorazrr) [reply src/pages/company/[id].tsx:51]: Let's move this below the tooltips!

File: src/components/Accordions/CompanyDetailsAccordion.tsx [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+const CompanyDetailsAccordion: FC<CompanyDetailsAccordionProps> = ({
+  content,
+}) => {
+  const [folded, setFolded] = useState(true)
+
+  const maxWordCount = 75
+
+  let truncated = content
+  const contentLengthTooLong = (content.match(/ /g) || []).length > maxWordCount
```
Comment (zorazrr) [inline src/components/Accordions/CompanyDetailsAccordion.tsx:18]: Nit. Could we do something like `const contentLengthTooLong = content.split(' ').length > maxWordCount`? Seems to be a bit more intuitive than `.match(/ /g)` and also aligns with what was used in `truncated` on line 20. We can even store the split result into an array and re-use it in both places. This also works though it's totally just a nit and up to you.

File: src/components/Accordions/CompanyDetailsAccordion.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  if (contentLengthTooLong) {
+    truncated = content.split(' ').slice(0, maxWordCount).join(' ')
+    truncated += '...'
+    console.log('truncated to ', truncated)
+  }
+
+  return (
+    <div className="mb-10 rounded-[10px] border-[3px]  border-cobalt bg-lightBlue px-10 py-5">
```
Comment (zorazrr) [inline src/components/Accordions/CompanyDetailsAccordion.tsx:26]:
```suggestion
    <div className="mb-10 rounded-xl border-[3px]  border-cobalt bg-lightBlue px-10 py-5">
```

**Review by zorazrr — CHANGES_REQUESTED** [review]
Nice work! Just some small nits and clean-ups.

File: src/components/Accordions/CompanyDetailsAccordion.tsx (hunk 3 above)
Comment (lizageorge) [reply src/components/Accordions/CompanyDetailsAccordion.tsx:26]: I think the exact radius from Figma was 10px, which was between `rounded-lg` and `rounded-xl`. But if we're sticking to TW standards instead I can make this change.

**Review by zorazrr — APPROVED** [review]
LGTM

**Pull request merged**